cli/src/common/boto3/ec2.py

```diff
@@ -179,6 +178,11 @@ def get_official_image_id(self, os, architecture, filters=None):
             raise AWSClientError(function_name="describe_images", message="Cannot find official ParallelCluster AMI")
         return max(images, key=lambda image: image["CreationDate"]).get("ImageId")
 
+    @AWSExceptionHandler.handle_client_exception
+    def get_eip_allocation_id(self, eip):
+        """Retrieve the allocation id of an Elastic IP."""
+        return self._client.describe_addresses(PublicIps=[eip])["Addresses"][0]["AllocationId"]
```

Comment:
Should I handle the exception here? Or should I add an validator in the config layer?

Reply:
I'd say both. this code must be reliable by itself but a validator would be good to fail at validation time rather than a cluster creation time.

Reply:
I added a validator. And manually tried this method. It will throw an exception if the eip doesn't exist. The message is like `Error during execution of get_eip_allocation_id. Address '1.2.3.4' not found.. Error code: InvalidAddress.NotFound.`